cmd/controller-manager/duplicate_detector.go

```diff
@@ -0,0 +1,150 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/go-logr/logr"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/rest"
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	ctrlutil "github.com/kubestellar/kubestellar/pkg/ctrlutil"
+	kfv1alpha1 "github.com/kubestellar/kubeflex/api/v1alpha1"
+	kslclient "github.com/kubestellar/kubestellar/pkg/client"
+)
+
+type ObjectIdentifier struct {
+	Group     string
+	Version   string
+	Kind      string
+	Name      string
+	Namespace string
+}
+
+// DetectDuplicateObjectIdentifiersAcrossWDSes checks for duplicate object identifiers across all WDSes and logs warnings if found.
+func DetectDuplicateObjectIdentifiersAcrossWDSes(logger logr.Logger) error {
+	ctx := context.Background()
```

Comment:
The usual pattern is to pass the `ctx context.Context` as a parameter and then use `klog.FromContext` to extract the logger from `ctx`.